### PR TITLE
#177: Add test for duplicate instance name

### LINKED ISSUE #177: Expose validation error when creating extension instance
When we create an extension instance and there is a validation issue coming from the extension, we currently cannot inform the user about it since EM doesn't return e.g. a bad request error for it.
Example error message returned:
```
failed to add instance for extension "row-level-security-lua": error executing statement 'CREATE VIRTUAL SCHEMA "v_schema_name" USING "EXA_EXTENSIONS"."RLS_ADAPTER" WITH SCHEMA_NAME = 'Schema Name' LOG_LEVEL='DEBUG'': E-EGOD-11: execution failed with SQL error code '42500' and message 'schema "v_schema_name" already exists (Session: 1797399225720176641)'
```

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #177

This adds a shared integration test for checking the error message when a virtual schema already exists.

### CODE REVIEW

File: extension-manager-integration-test-java/src/main/java/com/exasol/extensionmanager/itest/ExasolMetadata.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public void assertScript(final Matcher<ResultSet> matcher) {
                         + " WHERE SCRIPT_SCHEMA=?" //
                         + " ORDER BY SCRIPT_NAME")) {
             statement.setString(1, this.extensionSchemaName);
-            assertThat(statement.executeQuery(), matcher);
+            assertThat("SYS.EXA_ALL_SCRIPTS content", statement.executeQuery(), matcher);
```
Comment (tomuben) [inline extension-manager-integration-test-java/src/main/java/com/exasol/extensionmanager/itest/ExasolMetadata.java:38]: Hm, shouldn't be also `assertResult`, instead of `assertThat`?

File: extension-manager-integration-test-java/src/main/java/com/exasol/extensionmanager/itest/base/AbstractVirtualSchemaExtensionIT.java [hunk 2]
```diff
@@ -287,14 +292,25 @@ private PreviousExtensionVersion createPreviousVersion() {
...
+            final PreviousExtensionVersion previousVersion) {
         // The extension is installed twice (previous and current version), so each one returns one installation.
-        assertThat(getSetup().client().getInstallations(),
-                containsInAnyOrder(
-                        new InstallationsResponseInstallation().name(config.getExtensionName()).version(expectedVersion)
-                                .id(config.getExtensionId()), //
-                        new InstallationsResponseInstallation().name(config.getExtensionName()).version(expectedVersion)
-                                .id(previousVersion.getExtensionId())));
+        final List<InstallationsResponseInstallation> installations = getSetup().client().getInstallations();
+        assertAll(() -> assertThat("installations", installations, hasSize(2)),
```
Comment (tomuben) [inline extension-manager-integration-test-java/src/main/java/com/exasol/extensionmanager/itest/base/AbstractVirtualSchemaExtensionIT.java:307]:
```suggestion
        assertAll(() -> assertThat("number installations", installations, hasSize(2)),
```

File: doc/changes/changes_0.5.11.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# Extension Manager 0.5.11, released 2024-05-07
+
+Code name: Improve error message for creating duplicate instances
+
+## Summary
+
+This release adds a new test to the shared integration test class `AbstractVirtualSchemaExtensionIT` that verifies the error message in case the user tries to create a virtual schema that already exists. It also allows overriding the parameter name of the virtual schema. This is required in case an extensions is not based on the base virtual schema extension.
```
Comment (ckunki) [inline doc/changes/changes_0.5.11.md:7]:
```suggestion
This release adds a new test to the shared integration test class `AbstractVirtualSchemaExtensionIT` verifying the error message in case the user tries to create a virtual schema that already exists. It also allows overriding the parameter name of the virtual schema. This is required in case an extension supports this parameter while not inheriting from the base virtual schema extension.
```

File: extension-manager-integration-test-java/src/main/java/com/exasol/extensionmanager/itest/base/AbstractVirtualSchemaExtensionIT.java [hunk 4]
```diff
@@ -308,6 +324,30 @@ public void virtualSchemaWorks() {
...
+     */
+    @Test
+    public void createVirtualSchemaWithDuplicateNameFails() {
+        getSetup().client().install();
+        prepareInstance();
+        createInstance("my_VS");
+        getSetup().client().assertRequestFails(() -> createInstance("my_VS"), //
+                "Virtual Schema 'my_VS' already exists", 400);
+    }
+
```
Comment (ckunki) [inline extension-manager-integration-test-java/src/main/java/com/exasol/extensionmanager/itest/base/AbstractVirtualSchemaExtensionIT.java:338]: Maybe the case-insensitive case would be sufficient ....

**Review by ckunki — CHANGES_REQUESTED** [review]

File: extension-manager-integration-test-java/src/main/java/com/exasol/extensionmanager/itest/base/AbstractVirtualSchemaExtensionIT.java (hunk 4 above)
Comment (kaklakariada) [reply extension-manager-integration-test-java/src/main/java/com/exasol/extensionmanager/itest/base/AbstractVirtualSchemaExtensionIT.java:338]: You are right. This will speed up tests!

**Review by ckunki — APPROVED** [review]
